sopel/logger.py

```diff
@@ -33,12 +31,22 @@ def formatException(self, exc_info):
 
 
 def setup_logging(bot):
-    level = bot.config.core.logging_level or 'WARNING'
-    logging.basicConfig(level=level)
+    params = {'level': bot.config.core.logging_level or 'WARNING'}
+    if bot.config.core.logging_format:
+        params['format'] = bot.config.core.logging_format
+    if bot.config.core.logging_datefmt:
+        params['datefmt'] = bot.config.core.logging_datefmt
+    logging.basicConfig(**params)
     logger = logging.getLogger('sopel')
     if bot.config.core.logging_channel:
+        level = (bot.config.core.logging_channel_level or
+                 bot.config.core.logging_level or 'WARNING')
```

Comment:
I think you should define a variable `base_level = bot.config.core.logging_level or 'WARNING'`, and the use it everywhere you need it:

```
base_level = bot.config.core.logging_level or 'WARNING'
params = {'level': base_level}

# ...
level = (bot.config.core.logging_channel_level or base_level)
```

Reply:
And same for `bot.config.core.logging_format` with `bot.config.core.logging_channel_format` btw.

Reply:
just updated to use base_level et. al. as suggested; I also added channel_* vars and a channel_params dict, so only those params that are defined are included in the formatter contstructor.